### DISCUSSION TITLE
Refresh token rotation - jwt callback called multiple times

### DISCUSSION THREAD

**Discussion by barbara-lea** [nextauthjs/next-auth#1984] [Help] [answered]
Hi :) 

While trying to add a refresh token mechanism based on the [provided example](https://next-auth.js.org/tutorials/refresh-token-rotation) I face some errors which I don't know how to overcome. The `refreshAccessToken` function gets called multiple times with the same token, resulting one time with a successful response from the custom OAuth provider and after that in some errors stating that "refresh token is invalid". The error makes sense, cause the same refresh token should only be used once, but I don't know how to make the `jwt` callback only fire once, so it won't try to call the `refreshAccessToken` multiple times.   
What I am suspecting that complicates things is that my SWR's fetcher function calls `getSession()` in order to get the `apiToken` and include it in the request headers.  
Maybe it's worth mentioning that the setup in the `/api/auth/[...nextauth].ts` is almost identical to the one in the example. 

Does anyone have some tips on what I may be doing wrong or what I am missing? Thank you so much.

**Comment by balazsorban44** [comment, marked as answer]
I think this discussion is a duplicate of https://github.com/nextauthjs/next-auth/discussions/1455, you could follow that one maybe. Unfortunately, we don't have a solution yet

**Reply by barbara-lea** [reply]
Yes, it seems that my use case is very similar to the one mentioned. I'll follow that, thanks for your response.
